src/hint/Hint.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react'
+import cc from 'classcat'
+import uniqueid from 'lodash.uniqueid'
+
+import { A11yProps } from '_utils/interfaces'
+import HintBubble, { HintBubblePosition } from './HintBubble'
+
+export interface HintProps extends A11yProps {
+  children: (id: string) => React.ReactNode
+  title: string
+  closeButtonTitle?: string
+  description?: string
+  position?: HintBubblePosition
+  className?: Classcat.Class
+  onClose?: () => void
+  hidden?: boolean
+}
+
+export interface HintState {
+  hiddenBubble: boolean
+}
+
+class Hint extends Component<HintProps, HintState> {
+  static defaultProps: Partial<HintProps> = {
+    description: null,
+    closeButtonTitle: null,
+    position: HintBubblePosition.ABOVE,
+    className: '',
+    onClose() {},
+    hidden: false,
+  }
+
+  state: HintState = {
+    hiddenBubble: this.props.hidden,
+  }
+
+  id: string = ''
+
+  constructor(props: HintProps) {
+    super(props)
+    this.id = uniqueid('kirk-hint-')
+  }
+
+  onClose = () => {
+    this.setState({ hiddenBubble: true })
+  }
+
+  render() {
+    const {
+      className,
+      title,
+      description,
+      position,
+      children,
+      closeButtonTitle,
+      onClose,
+      hidden,
+      ...props
+    } = this.props
+    return (
+      <div className={cc([className, { 'hidden-bubble': this.state.hiddenBubble }])} {...props}>
+        {children(this.id)}
```

Comment:
Visually, the bubble appears first (above the hiint target). We should have it first in source order too.

Reply:
The aria attribute will make the link between the two components for screen readers. When reading, I think it's better to have first the main content, and then we have the additional informations on it. And I don't want the visual position of the bubble to impact the DOM reading flow.
Maybe @nickoola can confirm or infirm this.

Reply:
In terms on reading order, we should have the Hint content before. Here it's like a `tips label/title`. More over the tabbing order is probably broken because of that (Bubble >> Target >> Close) instead of (Bubble >> Close >> Target).

Reply:
After screen reader + keyboard navigation tests, I confirm that moving `{children(this.id)}` after the `HintBubble ` will fix the reading order. To resume the appropriate order should be **Hint content >> Hint Close >> The target**